src/client.ts

```diff
@@ -0,0 +1,222 @@
+export interface AuthorizeArgs {
+  /**
+   * The optional message to provide to the extension
+   */
+  message?: string;
+}
+
+export interface AuthorizeResultCredential {
+  /**
+   * The credential data
+   */
+  raw: unknown;
+
+  /**
+   * The credential data as a CESR encoded key event stream
+   */
+  cesr: string;
+}
+
+export interface AuthorizeResult {
+  /**
+   * If the extension responds with a credential, the data will be contained here.
+   */
+  credential?: AuthorizeResultCredential;
+
+  /**
+   * The session id of this authorization, if permissions to sign headers were requested, this
+   * session id needs to be provided.
+   */
+  sessionId: string;
```

Comment:
@lenkan I would appreciate it if you could explain how is this implemented this sessionId in your extension. 
Is it made by combining an identifier for the tab e.g sender.tab.id? 
Or it works if web app is opened in multiple tabs? It would be good for us to implement this in signify-browser-extension.

Reply:
Sorry for the delay @HunnySajid. I have just gotten back to this now and I think the sessionId string should actually be removed from the API here. The purpose of it was to be able to handle multiple tabs/sessions at the same time but isolated from each other. 

However, I think the solution is a bit flawed. I would rather do what you suggested with `sender.tab.id` to keep track of the sender. In that implementation, the session id does not have to be part of the polaris API. It would just be an implementation detail of the browser extension. I am going to try and make those changes ASAP.

Reply:
Thanks @lenkan 
this way the session is associated with a distinguish tab instead of a webapp (or domain if you will). No need to share sessionId with webapp.

Reply:
@HunnySajid I have updated the API now to remove the "sessionId" parameter.

I believe the things we need to land this is to re-introduce compatibility with the signify browser extension. However, for that, I think we either need to do some sort of "handshake" to determine if "externally_connectable" is enabled, or change the implementation in signify-browser-extension to based on post message for all features.

Reply:
> @HunnySajid I have updated the API now to remove the "sessionId" parameter.
> 
> I believe the things we need to land this is to re-introduce compatibility with the signify browser extension. However, for that, I think we either need to do some sort of "handshake" to determine if "externally_connectable" is enabled, or change the implementation in signify-browser-extension to based on post message for all features.

@lenkan thank you for making the relevant changes. I have been lately busy with compatible changes on signify-browser-extension side (was working with [source branch](https://github.com/nordlei/polaris-web/tree/signed-headers) of this pr). The externally_connetable is omitted now, postmessage is used instead. As soon as this PR is merged i'll create a couple of subsequent PRs on browser extension side for compatibility.